public/app/features/alerting/unified/hooks/useFilteredRules.ts

```diff
@@ -183,6 +188,15 @@ const reduceGroups = (filterState: RulesFilter) => {
       filteredRules = fuzzyFilter(filteredRules, (r) => r.name, ruleNameQuery);
     }
 
+    // Filter by rule source at rule-level (Grafana-managed vs datasource-managed)
+    if (filterState.ruleSource) {
+      const grafanaSelected = filterState.ruleSource === RuleSource.Grafana;
+      filteredRules = filteredRules.filter((rule) => {
+        const isGrafana = !!(rule.rulerRule && rulerRuleType.grafana.rule(rule.rulerRule));
+        return grafanaSelected ? isGrafana : !isGrafana;
```

Comment:
nit: `return grafanaSelected && isGrafana` makes more sense to me when trying to read this – personal preference though so feel free to keep as-is :D